posthog/models/event/util.py

```diff
@@ -87,31 +51,42 @@ def create_event(
         "uuid": str(event_uuid),
         "event": event,
         "properties": json.dumps(properties),
-        "timestamp": timestamp.strftime("%Y-%m-%d %H:%M:%S.%f"),
+        "timestamp": format_clickhouse_timestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S.%f"),
         "team_id": team.pk,
         "distinct_id": str(distinct_id),
         "elements_chain": elements_chain,
         "created_at": timestamp.strftime("%Y-%m-%d %H:%M:%S.%f"),
         "person_id": str(person_id) if person_id else "00000000-0000-0000-0000-000000000000",
         "person_properties": json.dumps(person_properties) if person_properties is not None else "",
-        "person_created_at": person_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if person_created_at is not None else 0,
+        "person_created_at": format_clickhouse_timestamp(person_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f")
+        if person_created_at is not None
+        else 0,
         "group0_properties": json.dumps(group0_properties) if group0_properties is not None else "",
         "group1_properties": json.dumps(group1_properties) if group1_properties is not None else "",
         "group2_properties": json.dumps(group2_properties) if group2_properties is not None else "",
         "group3_properties": json.dumps(group3_properties) if group3_properties is not None else "",
         "group4_properties": json.dumps(group4_properties) if group4_properties is not None else "",
-        "group0_created_at": group0_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if group0_created_at is not None else 0,
-        "group1_created_at": group1_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if group1_created_at is not None else 0,
-        "group2_created_at": group2_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if group2_created_at is not None else 0,
-        "group3_created_at": group3_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if group3_created_at is not None else 0,
-        "group4_created_at": group4_created_at.strftime("%Y-%m-%d %H:%M:%S.%f") if group4_created_at is not None else 0,
+        "group0_created_at": format_clickhouse_timestamp(group0_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f"),
+        "group1_created_at": format_clickhouse_timestamp(group1_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f"),
+        "group2_created_at": format_clickhouse_timestamp(group2_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f"),
+        "group3_created_at": format_clickhouse_timestamp(group3_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f"),
+        "group4_created_at": format_clickhouse_timestamp(group4_created_at, ZERO_DATE).strftime("%Y-%m-%d %H:%M:%S.%f"),
     }
     p = ClickhouseProducer()
     p.produce(topic=KAFKA_EVENTS_JSON, sql=INSERT_EVENT_SQL(), data=data)
 
     return str(event_uuid)
 
 
+# default
+def format_clickhouse_timestamp(
```

Comment:
why not put `.strftime("%Y-%m-%d %H:%M:%S.%f")` in here too?

Reply:
🤦  yeah that makes sense too :D